File: 0535-encode-and-decode-tinyurl.java
/*
535. Encode and Decode TinyURL

Submitted: March 6, 2025

Runtime: 2 ms (beats 69.39%)
Memory: 43.03 MB (beats 55.63%)
*/

import java.util.HashMap;

public class Codec {

    private final HashMap<Integer, String> map = new HashMap<>();

    // Encodes a URL to a shortened URL.
    public String encode(String longUrl) {
        map.put(longUrl.hashCode(), longUrl);
        return Integer.toString(longUrl.hashCode());
    }

    // Decodes a shortened URL to its original URL.
    public String decode(String shortUrl) {
        return map.get(Integer.parseInt(shortUrl));
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(url));

File: 0535-encode-and-decode-tinyurl.cpp
/*
535. Encode and Decode TinyURL

Submitted: October 7, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 9.30 MB (beats 90.58%)
*/

class Solution {
public:

    // Encodes a URL to a shortened URL.
    const string& encode(const string& longUrl) {
        return longUrl;
    }

    // Decodes a shortened URL to its original URL.
    const string& decode(const string& shortUrl) {
        return shortUrl;
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));